bitcoin/src/blockdata/transaction.rs

```diff
@@ -829,6 +829,41 @@ impl Transaction {
     }
 }
 
+/// The transaction version.
+///
+/// We only support standard transaction versions, this is currently 1 or 2 as specified in [BIP-68]
+/// which introduced relative lock-time consensus-enforced semantics of the sequence number.
+///
+/// [BIP-68]: https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki
+#[derive(Copy, PartialEq, Eq, Clone, Debug, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Version(i32);
+
+impl Version {
+    /// The original Bitcoin transaction version (pre-BIP-68).
+    pub const ONE: Self = Self(1);
+
+    /// The second Bitcoin transaction version (post-BIP-68).
+    pub const TWO: Self = Self(2);
+}
+
+impl Default for Version {
+    fn default() -> Version { Version::TWO }
+}
+
+impl Encodable for Version {
+    fn consensus_encode<W: io::Write + ?Sized>(&self, w: &mut W) -> Result<usize, io::Error> {
+        self.0.consensus_encode(w)
+    }
+}
+
+impl Decodable for Version {
+    fn consensus_decode<R: io::Read + ?Sized>(r: &mut R) -> Result<Self, encode::Error> {
+        Decodable::consensus_decode(r).map(Version)
```

Comment:
Should this not check that these are the only values we support. I think we should allow for other values too, perhaps under another constructor. People are prototyping tx ver 3 in signet for mempool relay stuff.

Reply:
Just in case you missed it, current version of this PR includes `transaction::Version::non_standard` constructor as requested.